### DISCUSSION TITLE
No DRAM activity

### DISCUSSION THREAD

**Discussion by Sacusa** [accel-sim/accel-sim-framework#185] [Q&A]
I want to use GPGPU-Sim to collect memory statistics and am trying to run a small vector addition kernel using PTX simulation. After the application finishes execution, the output verification passes but there is no DRAM activity. I enabled the flushing of L2 cache after kernel completion, but that didn't help. Running the simulation in a debugger confirms that `dram_t::push` is indeed never called either. Does anyone know what's happening here?

**Comment by JRPan** [comment]
One reason could be your data is too small. By default, cudamemcpy loads into L2. So if your working set can fit in L2 then it's possible there is no memory traffic. You can try turning off `-gpgpu_perf_sim_memcpy`. But I'm not sure how much this would help. Please let us know!

Thanks

**Reply by Sacusa** [reply]
It didn't help, unfortunately. I'm curious why there is no memory traffic even for small kernels if the L2 cache is being flushed. Is there no way to generate memory traffic for such kernels?

**Reply by JRPan** [reply]
Apparently, L2 flush only reset the tag array. It does not write back to DRAM. It actually has a comment says `// TODO: write the flushed data to the main memory`, which I didn't know of lol. 
https://github.com/accel-sim/gpgpu-sim_distribution/blob/dev/src/gpgpu-sim/l2cache.cc#L703

Well I thought disabling the sim_memcpy would work. Maybe it's the pending request. I cannot really tell you exactly what the problem is. But I'm pretty sure there is no known bug related to that.

Thanks

**Reply by JRPan** [reply]
Actually, if you just want to see the memory traffic you can play with the cache configs. Disable caching completely. So every read and write goes to DRAM directly. That would probably work.

**Reply by Sacusa** [reply]
I'll try disabling the caches and post here, thanks!
